function/aggregate/aggregate.go

```diff
@@ -64,10 +88,11 @@ func addToGroup(rows []group, series api.Timeseries, tags []string) []group {
 }
 
 // groupBy groups the given SeriesList by tags, producing a list of lists (of type groupResult)
-func groupBy(list api.SeriesList, tags []string) []group {
+// if `combines` is true, then it groups on all other tags instead.
+func groupBy(list api.SeriesList, tags []string, combines bool) []group {
 	result := []group{}
 	for _, series := range list.Series {
-		result = addToGroup(result, series, tags)
+		result = addToGroup(result, series, tags, combines)
```

Comment:
I found a different way of organizing things, by changing `groupAccepts` to not need either `tags` or `combines`, and instead just operate on the tagsets.

There's still one `if collapses { ...` left behind, but there's only the one, and I manage to avoid having to pass functions as parameters.